internal/csi-addons/rbd/replication.go

```diff
@@ -937,3 +945,37 @@ func checkVolumeResyncStatus(ctx context.Context, localStatus types.SiteStatus)
 
 	return nil
 }
+
+func getCurrentReplicationStatus(ctx context.Context,
+	mirrorGlobalStatus types.GlobalStatus,
+) (replication.GetVolumeReplicationInfoResponse_Status, string, error) {
+	// get the state of replication
+	localStatus, err := mirrorGlobalStatus.GetLocalSiteStatus()
+	if err != nil {
+		log.ErrorLog(ctx, err.Error())
+
+		return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "", err
+	}
+
+	localState := localStatus.GetState()
+
+	// check if mirroring state is up/down
+	isUp := localStatus.IsUP()
+
+	if isUp {
+		switch localState {
+		case librbd.MirrorImageStatusStateUnknown.String():
+			return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "mirroring is in unknown state", nil
+		case librbd.MirrorImageStatusStateError.String():
+			return replication.GetVolumeReplicationInfoResponse_ERROR, "mirroring is in error state", nil
+		case librbd.MirrorImageStatusStateReplaying.String():
+			return replication.GetVolumeReplicationInfoResponse_HEALTHY, "image/group is secondary", nil
+		case librbd.MirrorImageStatusStateStopped.String():
+			return replication.GetVolumeReplicationInfoResponse_HEALTHY, "image/group is primary", nil
+		default:
+			return replication.GetVolumeReplicationInfoResponse_UNKNOWN, "mirroring is in intermediatory state", nil
```

Comment:
How about returning the message present in the description as it contains the more meaningful details required for debugging. passing own message might not be helpful here.

Reply:
Ack, that can be done.
Though, only for the case of secondary I was thinking of passing it manually as the description for a secondary cluster is only "replaying", wdyt?

Reply:
Having native details returned from ceph helps in debugging. Lets add whatever the details we get from ceph , when we dont get anything we can have ceph-csi specific generic response.